internal/lookout/repository/jobs.go

```diff
@@ -248,6 +252,31 @@ func sortJobsByJobId(jobInfos []*lookout.JobInfo, descending bool) {
 	})
 }
 
+func determineJobStateDuration(row *JobRow) string {
+	if row == nil {
+		return ""
+	}
+	var timeStamp time.Time
+
+	switch {
+	case row.Finished.Valid:
+		timeStamp = row.Finished.Time
+	case row.Started.Valid:
+		timeStamp = row.Started.Time
+	case row.Created.Valid:
+		timeStamp = row.Created.Time
+	case row.Cancelled.Valid:
+		timeStamp = row.Cancelled.Time
+	case row.Submitted.Valid:
```

Comment:
what about other states like PENDING or Queued

Reply:
I believe the other states are captured here -- if nothing but "submitted" has a value, it is queued, etc

Reply:
Don’t we want time in queued or pending?

This looks to map state based on Time.
https://github.com/G-Research/armada/blob/master/internal/lookout/repository/jobs.go

Reply:
This could be right but I find it confusing.  Started and Created don't match the states in the lookout code.  I think Started corresponds to Running and Created would correspond to Queued.  But could you use the JobState as the key for the switch instead to make this a bit more clear

Reply:
Looks good!